src/main/java/seedu/nutrigoals/logic/commands/FindCommand.java

```diff
@@ -26,22 +29,40 @@ public class FindCommand extends Command {
     public static final String MESSAGE_FOOD_CALORIES_NOT_FOUND = "No calorie content information for %1$s";
 
     private final Name foodName;
+    private FilteredList<Food> filteredList;
+    private final IsFoodInPredicate predicate;
 
     /**
      * Creates a FindCommand to find the calorie content of the given food.
      * @param foodName The {@code Name} of the food to find the calorie content of.
      */
     public FindCommand(Name foodName) {
         this.foodName = foodName;
+        predicate = new IsFoodInPredicate(foodName);
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         Map<Name, Calorie> foodCalorieList = model.getFoodCalorieList();
+        filteredList = new FilteredList<>(model.getUnFilteredFoodList());
+        filteredList.setPredicate(predicate);
+
+        // if food is in filtered list return average of calories in list
+        if (filteredList.size() != 0) {
+            Integer sum = 0;
+            for (Food food : filteredList) {
+                sum += food.getCalorie().getCalorieValue();
+            }
+            sum /= filteredList.size();
+            Calorie average = new Calorie(sum.toString());
+            return new CommandResult(String.format(MESSAGE_FIND_FOOD_CALORIE_SUCCESS, foodName, average));
+        }
+
         if (!foodCalorieList.containsKey(foodName)) {
```

Comment:
`rice` can be found in the `foodCalorieList` with 260 calories, and what if the user keys in the following commands:

```
add n/rice t/lunch c/300
find rice
```

Should the command return 300 calories or (260 + 300) / 2?

Reply:
Current implementation is such that if the food item can be found in the list that the user keys in, it would take priority over the default value. What do you think is better?